app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js

```diff
@@ -7,229 +7,98 @@ import {
-                  }
-
-                  get prefixType() {
-                    return "icon";
-                  }
-
-                  get prefixValue() {
-                    return "d-chat";
-                  }
-
-                  get prefixColor() {
-                    return "FF0000";
-                  }
-
-                  get prefixBadge() {
-                    return "lock";
-                  }
-
-                  get suffixType() {
-                    return "icon";
-                  }
-
-                  get suffixValue() {
-                    return "circle";
-                  }
-
-                  get suffixCSSClass() {
-                    return "unread";
-                  }
-
-                  @bind
-                  didInsert() {
-                    linkDidInsert = "link test";
-                  }
-
-                  @bind
-                  willDestroy() {
-                    linkDestroy = "link test";
-                  }
-                })(),
-                new (class extends BaseCustomSidebarSectionLink {
-                  get name() {
-                    return "dev-channel";
-                  }
-
-                  get route() {
-                    return "discovery.latest";
-                  }
-
-                  get title() {
-                    return "dev channel title";
-                  }
-
-                  get text() {
-                    return "dev channel text";
-                  }
-
-                  get prefixColor() {
-                    return "alert";
-                  }
-
-                  get prefixType() {
-                    return "text";
-                  }
-
-                  get prefixValue() {
-                    return "test text";
-                  }
-                })(),
-                new (class extends BaseCustomSidebarSectionLink {
-                  get name() {
-                    return "fun-channel";
-                  }
-
-                  get route() {
-                    return "discovery.latest";
-                  }
-
-                  get title() {
-                    return "fun channel title";
-                  }
-
-                  get text() {
-                    return "fun channel text";
-                  }
-
-                  get prefixType() {
-                    return "image";
-                  }
-
-                  get prefixValue() {
-                    return "/test.png";
-                  }
-
-                  get hoverType() {
-                    return "icon";
-                  }
-
-                  get hoverValue() {
-                    return "xmark";
-                  }
-
-                  get hoverAction() {
-                    return () => {};
-                  }
-
-                  get hoverTitle() {
-                    return "hover button title attribute";
-                  }
-                })(),
+            name = "test-chat-channels";
```

Comment:
TIL that fields can override getters in the base class!

Reply:
yup! and we do use it already in a couple of places with the sidebar api